magento/magento2
# Reuse password reset tokens until they expire

### Description (*)

This avoids a race condition where email delivery is slow and user generates
new token before original (now invalid) token is received.

Consider the following scenario:
1. User requests token (1)
1. User waits, but does not yet receive the token
1. User requests another token (2)
1. User receives token (1), but this is invalid
1. User requests another token (3)
1. User receives token (2), but this is invalid
1. [repeat]

This also has the added bonus of not giving an attacker sequential tokens upon request. While the tokens are understood to be suitably opaque, any information disclosures identified in future will be at least partially mitigated by reusing the same token rather than giving out another token data point.

### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Generate password reset email (admin & frontend)
1. Wait for time-out
1. Generate another password reset email
1. Observe same token in both emails

### Questions or comments

I am unsure if this change requires a unit test.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#29640: Reuse password reset tokens until they expire
